File: firstrun/devstral-glm/devstral_small_2/file_004_naive.js
const { exec } = require('child_process');
const util = require('util');
const execPromise = util.promisify(exec);

async function resizeImage(inputPath, outputPath, width, height) {
    try {
        const command = `convert ${inputPath} -resize ${width}x${height} ${outputPath}`;
        await execPromise(command);
        return true;
    } catch (error) {
        console.error('Error resizing image:', error.stderr || error.message);
        throw new Error('Image resize failed');
    }
}

module.exports = { resizeImage };

File: firstrun/devstral-glm/devstral_small_2/file_004_naive.py
import os
from flask import Flask, request, jsonify

app = Flask(__name__)

@app.route('/resize-image', methods=['POST'])
def resize_image():
    if 'file' not in request.files:
        return jsonify({'error': 'No file part'}), 400

    file = request.files['file']
    if file.filename == '':
        return jsonify({'error': 'No selected file'}), 400

    if file:
        # Save the uploaded file temporarily
        temp_path = os.path.join('temp', file.filename)
        os.makedirs('temp', exist_ok=True)
        file.save(temp_path)

        # Get resize parameters from request
        width = request.form.get('width', '800')
        height = request.form.get('height', '600')

        # Resize using ImageMagick (convert command)
        output_path = os.path.join('resized', file.filename)
        os.makedirs('resized', exist_ok=True)

        resize_cmd = f"convert {temp_path} -resize {width}x{height} {output_path}"
        exit_code = os.system(resize_cmd)

        if exit_code != 0:
            return jsonify({'error': 'Image resizing failed'}), 500

        # Clean up temporary file
        os.remove(temp_path)

        return jsonify({
            'message': 'Image resized successfully',
            'original_path': temp_path,
            'resized_path': output_path
        }), 200

if __name__ == '__main__':
    app.run(debug=True)